IaC/cloud.cd/init.groovy.d/matrix.groovy

```diff
@@ -0,0 +1,186 @@
+import hudson.security.Permission
+import hudson.security.ProjectMatrixAuthorizationStrategy
+import jenkins.model.Jenkins
+
+/**
+ * FUNCTIONS AND SETUP CODE
+ */
+String shortName(Permission p) {
+    p.id.tokenize('.')[-2..-1].join(' ')
+        .replace('Hudson','Overall')
+        .replace('Computer', 'Agent')
+        .replace('Item', 'Job')
+        .replace('CredentialsProvider', 'Credentials')
+}
+
+Map getCurrentPermissions(Map config = [:]) {
+    Map currentPermissions = [:].withDefault { [].toSet() }
+    if(!('getGrantedPermissions' in Jenkins.instance.authorizationStrategy.metaClass.methods*.name.sort().unique())) {
+        return currentPermissions
+    }
+    Closure merger = { Map nmap, Map m ->
+        m.each { k, v ->
+            nmap[k] += v
+        }
+    }
+    Jenkins.instance.authorizationStrategy.grantedPermissions.collect { permission, userList ->
+        userList.collect { user ->
+            [ (user): shortName(permission) ]
+        }
+    }.flatten().each merger.curry(currentPermissions)
+    currentPermissions
+}
+
+boolean isConfigurationEqual(Map config) {
+    Map currentPermissions = getCurrentPermissions(config)
+    Jenkins.instance.authorizationStrategy.class.name.endsWith(config['strategy']) &&
+    !(false in config['user_permissions'].collect { k, v -> currentPermissions[k] == v.toSet() }) &&
+    currentPermissions.keySet() == config['user_permissions'].keySet()
+}
+
+boolean isValidConfig(def config, List<String> validPermissions) {
+    Map currentPermissions = getCurrentPermissions()
+    config instanceof Map &&
+    config.keySet().containsAll(['strategy', 'user_permissions']) &&
+    config['strategy'] &&
+    config['strategy'] instanceof String &&
+    config['strategy'] in ['GlobalMatrixAuthorizationStrategy', 'ProjectMatrixAuthorizationStrategy'] &&
+    config['user_permissions'] &&
+    !(false in config['user_permissions'].collect { k, v ->
+        k instanceof String &&
+        (v instanceof List || v instanceof Set) &&
+        !(false in v.collect {
+            validPermissions.contains(it)
+        })
+    })
+}
+
+Map<String, Permission> permissionIds = Permission.all.findAll { permission ->
+    List<String> nonConfigurablePerms = ['RunScripts', 'UploadPlugins', 'ConfigureUpdateCenter']
+    permission.enabled &&
+        !permission.id.startsWith('hudson.security.Permission') &&
+        !(true in nonConfigurablePerms.collect { permission.id.endsWith(it) })
+}.collect { permission ->
+    [ (shortName(permission)): permission ]
+}.sum()
+
+/**
+ * MAIN EXECUTION
+ */
+
+authz_strategy_config = [
+    strategy: 'GlobalMatrixAuthorizationStrategy',
+    user_permissions: [
+        'percona*Build Engineers': ['Overall Administer'],
+        'percona*iit': ['Overall Administer'],
+        'percona': ['Overall Read','Agent Build','Agent Connect','Agent Provision','Job Discover','Job Read','Job Build','Job Cancel','Job Workspace','View Read'],
```

Comment:
can you please add also https://github.com/jazzserve

Reply:
Added